### PR TITLE
ADEN-4358 | add possibility to force country code through URL

### PR DESCRIPTION
**Pull request by bognix**

### CODE REVIEW

File: extensions/wikia/AdEngine/AdEngine2Hooks.class.php [hunk 1]
```diff
@@ -94,6 +94,7 @@ public static function onInstantGlobalsGetVariables( array &$vars )
 	 */
 	public static function onWikiaSkinTopScripts( &$vars, &$scripts ) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/AdEngine2Hooks.class.php:95]: MAJOR Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 12 issues
* MAJOR 5 major
* MINOR 3 minor
* INFO 4 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [AdEngine2Hooks.class.php#L23](https://github.com/Wikia/app/blob/b78983aea4970e08bb7910111d583dceb7358eca/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L23): Remove the unused function parameters "$title, $article, $output, $user, $wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L200](https://github.com/Wikia/app/blob/b78983aea4970e08bb7910111d583dceb7358eca/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L200): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L225](https://github.com/Wikia/app/blob/b78983aea4970e08bb7910111d583dceb7358eca/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L225): Remove the unused function parameters "$jsExtensionPackages, $scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [geo.js#L144](https://github.com/Wikia/app/blob/b78983aea4970e08bb7910111d583dceb7358eca/resources/wikia/modules/geo.js#L144): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. MINOR [AdEngine2ContextService.class.php#L22](https://github.com/Wikia/app/blob/b78983aea4970e08bb7910111d583dceb7358eca/extensions/wikia/AdEngine/AdEngine2ContextService.class.php#L22): Define a constant instead of duplicating this literal "scripts" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [AdEngine2ContextService.class.php#L94](https://github.com/Wikia/app/blob/b78983aea4970e08bb7910111d583dceb7358eca/extensions/wikia/AdEngine/AdEngine2ContextService.class.php#L94): Remove the literal "false" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)
1. MINOR [AdEngine2ContextService.class.php#L96](https://github.com/Wikia/app/blob/b78983aea4970e08bb7910111d583dceb7358eca/extensions/wikia/AdEngine/AdEngine2ContextService.class.php#L96): Remove the literal "false" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)
1. INFO [AdEngine2ContextService.class.php#L47](https://github.com/Wikia/app/blob/b78983aea4970e08bb7910111d583dceb7358eca/extensions/wikia/AdEngine/AdEngine2ContextService.class.php#L47): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. INFO [AdEngine2ContextService.class.php#L59](https://github.com/Wikia/app/blob/b78983aea4970e08bb7910111d583dceb7358eca/extensions/wikia/AdEngine/AdEngine2ContextService.class.php#L59): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. INFO [AdEngine2Hooks.class.php#L26](https://github.com/Wikia/app/blob/b78983aea4970e08bb7910111d583dceb7358eca/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L26): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* MAJOR 5 major
* INFO 2 info

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [AdEngine2Hooks.class.php#L24](https://github.com/Wikia/app/blob/c1902ad7f5f3d363d8d0876f23d1f2907c872a07/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L24): Remove the unused function parameters "$title, $article, $output, $user, $wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L207](https://github.com/Wikia/app/blob/c1902ad7f5f3d363d8d0876f23d1f2907c872a07/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L207): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L232](https://github.com/Wikia/app/blob/c1902ad7f5f3d363d8d0876f23d1f2907c872a07/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L232): Remove the unused function parameters "$jsExtensionPackages, $scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [geo.js#L149](https://github.com/Wikia/app/blob/c1902ad7f5f3d363d8d0876f23d1f2907c872a07/resources/wikia/modules/geo.js#L149): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. INFO [AdEngine2Hooks.class.php#L27](https://github.com/Wikia/app/blob/c1902ad7f5f3d363d8d0876f23d1f2907c872a07/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L27): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)
1. INFO [geo.js#L169](https://github.com/Wikia/app/blob/c1902ad7f5f3d363d8d0876f23d1f2907c872a07/resources/wikia/modules/geo.js#L169): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* MAJOR 5 major
* INFO 2 info

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [AdEngine2Hooks.class.php#L24](https://github.com/Wikia/app/blob/fe46101f4d98c32641188a0e51c71365e9039753/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L24): Remove the unused function parameters "$title, $article, $output, $user, $wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L207](https://github.com/Wikia/app/blob/fe46101f4d98c32641188a0e51c71365e9039753/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L207): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L232](https://github.com/Wikia/app/blob/fe46101f4d98c32641188a0e51c71365e9039753/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L232): Remove the unused function parameters "$jsExtensionPackages, $scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [geo.js#L149](https://github.com/Wikia/app/blob/fe46101f4d98c32641188a0e51c71365e9039753/resources/wikia/modules/geo.js#L149): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. INFO [AdEngine2Hooks.class.php#L27](https://github.com/Wikia/app/blob/fe46101f4d98c32641188a0e51c71365e9039753/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L27): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)
1. INFO [geo.js#L169](https://github.com/Wikia/app/blob/fe46101f4d98c32641188a0e51c71365e9039753/resources/wikia/modules/geo.js#L169): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

**Review by fraszczakszymon — APPROVED** [review]

File: resources/wikia/modules/geo.js [hunk 2]
```diff
@@ -164,11 +172,11 @@
 	//TODO: Can we remove the double alias in window.Geo and Wikia.geo
```
Comment (WikiaTech) [inline resources/wikia/modules/geo.js:172]: INFO Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* MAJOR 5 major
* INFO 2 info

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [AdEngine2Hooks.class.php#L24](https://github.com/Wikia/app/blob/dbae64139473c9bf6cd2306e658f0f08e44c246a/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L24): Remove the unused function parameters "$title, $article, $output, $user, $wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L200](https://github.com/Wikia/app/blob/dbae64139473c9bf6cd2306e658f0f08e44c246a/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L200): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L225](https://github.com/Wikia/app/blob/dbae64139473c9bf6cd2306e658f0f08e44c246a/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L225): Remove the unused function parameters "$jsExtensionPackages, $scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [geo.js#L152](https://github.com/Wikia/app/blob/dbae64139473c9bf6cd2306e658f0f08e44c246a/resources/wikia/modules/geo.js#L152): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. INFO [AdEngine2Hooks.class.php#L27](https://github.com/Wikia/app/blob/dbae64139473c9bf6cd2306e658f0f08e44c246a/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L27): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* MAJOR 5 major
* INFO 2 info

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [AdEngine2Hooks.class.php#L24](https://github.com/Wikia/app/blob/77ba4d8f458ee3e134de4269f55fafe22b547fbb/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L24): Remove the unused function parameters "$title, $article, $output, $user, $wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L200](https://github.com/Wikia/app/blob/77ba4d8f458ee3e134de4269f55fafe22b547fbb/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L200): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L225](https://github.com/Wikia/app/blob/77ba4d8f458ee3e134de4269f55fafe22b547fbb/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L225): Remove the unused function parameters "$jsExtensionPackages, $scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [geo.js#L152](https://github.com/Wikia/app/blob/77ba4d8f458ee3e134de4269f55fafe22b547fbb/resources/wikia/modules/geo.js#L152): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. INFO [AdEngine2Hooks.class.php#L27](https://github.com/Wikia/app/blob/77ba4d8f458ee3e134de4269f55fafe22b547fbb/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L27): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

File: resources/wikia/modules/spec/geo.spec.js [hunk 3]
```diff
@@ -81,4 +93,10 @@ describe('Geo', function () {
 		expect(geo.isProperGeo(['XX', 'non-XX-SA'])).toBeTruthy();
 		expect(geo.isProperGeo(['XX', 'non-XX-EU'])).toBeFalsy();
 	});
+
+	it('returns country when forcedCountry cookie set', function () {
```
Comment (nandy-andy) [inline resources/wikia/modules/spec/geo.spec.js:97]: :heart: moar unit tests

**Review by nandy-andy — APPROVED** [review]

**Pull request merged**